### DISCUSSION TITLE
How to get rid of spacing at the bottom of a label

### DISCUSSION THREAD

**Discussion by j-jorge** [axmolengine/axmol#2040] [Q&A]
Hello, I have a label that uses the LukiestGuy font and as you can see in the capture below there is some space at the bottom of the glyphs (here we a semi-transparent red rectangle fitting the label node):

![image](https://github.com/user-attachments/assets/c8446b4d-86a7-4bf9-ac30-089535b4c37a)

I would like to get rid of the spacing on the last line as it makes it difficult to center the label, what are my options?

**Comment by rh101** [comment]
I don't recall there being a way to adjust any of that, and since the issue in the font, your best bet may be to edit the font file and adjust the specific aspects of the font that reduce or eliminate that gap.

Another option would be to figure out what percentage of the height is made of that gap, then apply it to whatever font size you use in order to vertically offset the position of the by half that gap size.  

If you need to add a background to the label, as shown in the sample image, then it may be easier to add the label as a child of a node that you use as a container, and then offset the label vertically within that node by half the gap length.  The node height should be label height minus gap length.  You can then color that node (or add a colored node the same size as it as a child etc.), and since the dimensions match the exact boundary of the label without the bottom gap, then it should give you the result you are after.  You then add that node to your scene, simply center aligning it vertically, which would show the label with the correct centering.

**Reply by j-jorge** [reply]
I managed to fix the font by tweaking the ascent and descent metrics with FontForge. Thanks :)
